rllib/models/experimental/tf/encoder.py

```diff
@@ -0,0 +1,158 @@
+
+
+class LSTMEncoder(Encoder, TFModel):
+    """An encoder that uses an LSTM cell and a linear layer."""
+
+    def __init__(self, config: ModelConfig) -> None:
+        Encoder.__init__(self, config)
+        TFModel.__init__(self, config)
+
+        self.lstm = nn.LSTM(
+            config.input_dim,
+            config.hidden_dim,
+            config.num_layers,
+            batch_first=config.batch_first,
+        )
+        self.linear = nn.Linear(config.hidden_dim, config.output_dim)
+
+    def get_initial_state(self):
+        config = self.config
+        return {
+            "h": torch.zeros(config.num_layers, config.hidden_dim),
+            "c": torch.zeros(config.num_layers, config.hidden_dim),
+        }
+
+    @property
+    def input_spec(self):
+        config = self.config
+        return SpecDict(
+            {
+                # bxt is just a name for better readability to indicated padded batch
+                SampleBatch.OBS: TFTensorSpecs("bxt, h", h=config.input_dim),
+                STATE_IN: {
+                    "h": TFTensorSpecs(
+                        "b, l, h", h=config.hidden_dim, l=config.num_layers
+                    ),
+                    "c": TFTensorSpecs(
+                        "b, l, h", h=config.hidden_dim, l=config.num_layers
+                    ),
+                },
+                SampleBatch.SEQ_LENS: None,
+            }
+        )
+
+    @property
+    def output_spec(self):
+        config = self.config
+        return SpecDict(
+            {
+                ENCODER_OUT: TFTensorSpecs("bxt, h", h=config.output_dim),
+                STATE_OUT: {
+                    "h": TFTensorSpecs(
+                        "b, l, h", h=config.hidden_dim, l=config.num_layers
+                    ),
+                    "c": TFTensorSpecs(
+                        "b, l, h", h=config.hidden_dim, l=config.num_layers
+                    ),
+                },
+            }
+        )
+
+    @check_input_specs("input_spec", filter=True, cache=False)
+    @check_output_specs("output_spec", cache=False)
+    def __call__(self, inputs: TensorDict, **kwargs) -> ForwardOutputType:
+        x = inputs[SampleBatch.OBS]
+        states = inputs[STATE_IN]
+        # states are batch-first when coming in
+        states = tree.map_structure(lambda x: x.transpose(0, 1), states)
+
+        x = add_time_dimension(
+            x,
+            seq_lens=inputs[SampleBatch.SEQ_LENS],
+            framework="torch",
+            time_major=not self.config.batch_first,
+        )
+        states_o = {}
+        x, (states_o["h"], states_o["c"]) = self.lstm(x, (states["h"], states["c"]))
+
+        x = self.linear(x)
+        x = x.view(-1, x.shape[-1])
+
+        return {
+            ENCODER_OUT: x,
+            STATE_OUT: tree.map_structure(lambda x: x.transpose(0, 1), states_o),
+        }
+
+
+class IdentityEncoder(TFModel):
+    def _forward(self, inputs: TensorDict, **kwargs) -> ForwardOutputType:
```

Comment:
docstring missing.
Explain here that the identity encoder "doesn't do anything" :)

Reply:
and that it'll be a mere placeholder to avoid too many if-else-blocks in those algos/rl-modules that use encoders.

Reply:
Done!